File: doc-examples/mongo-example-kotlin/src/main/kotlin/example/Student.kt
package example

import io.micronaut.data.annotation.*
import org.bson.types.ObjectId

@MappedEntity
data class Student(
        @field:Id @GeneratedValue
        val id: ObjectId?,
        @field:Version
        val version: Long?,
        val name: String,
        @Relation(value = Relation.Kind.MANY_TO_MANY, cascade = [Relation.Cascade.PERSIST])
        val courses: List<Course>,
        @Relation(value = Relation.Kind.ONE_TO_MANY, mappedBy = "student")
        val ratings: List<CourseRating>
) {
    constructor(name: String, items: List<Course>) : this(null, null, name, items, emptyList())
}

File: doc-examples/mongo-example-python/src/main/python/example/Student.py
# tag::student[]
from dataclasses import dataclass
from typing import Annotated

from micronaut.data.annotation import GeneratedValue, Id, MappedEntity, Version
from org.bson.types import ObjectId


@MappedEntity
@dataclass
class Student:
    id: Annotated[ObjectId | None, Id, GeneratedValue] = None
    version: Annotated[int | None, Version] = None
    # end::student[]
    name: str | None = None
